src/main/webapp/resources/js/pages/announcement/components/AnnouncementsPage.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useState } from "react";
+import { PageWrapper } from "../../../components/page/PageWrapper";
+import {
+  getAnnouncements,
+  markAnnouncementRead,
+} from "../../../apis/announcements/announcements";
+import { Avatar, Form, List, notification, Radio, Typography } from "antd";
+import { PriorityFlag } from "./PriorityFlag";
+import { fromNow } from "../../../utilities/date-utilities";
+import ViewReadAnnouncement from "./ViewReadAnnouncement";
+import ViewUnreadAnnouncement from "./ViewUnreadAnnouncement";
+import { blue1 } from "../../../styles/colors";
+
+const { Text } = Typography;
+
+/**
+ * React component to display the page for administration of users.
+ * @returns {*}
+ * @constructor
+ */
+export function AnnouncementsPage({}) {
+  const [announcements, setAnnouncements] = useState([]);
+  const [readToggle, setReadToggle] = useState(false);
+  const [filter, setFilter] = useState("all");
+
+  useEffect(() => {
+    getAnnouncements().then((data) => {
+      switch (filter) {
+        case "read":
+          return setAnnouncements(
+            data.data.filter((item) => item.announcementUserJoin !== null)
+          );
+        case "unread":
+          return setAnnouncements(
+            data.data.filter((item) => item.announcementUserJoin === null)
+          );
+        default:
+          return setAnnouncements(data.data);
+      }
+    });
+  }, [filter, readToggle]);
+
+  function markAnnouncementAsRead(aID) {
+    return markAnnouncementRead({ aID })
```

Comment:
As I said for your other table, I think this is better down by just removing the announcement from the original list, instead of doing a full table refresh.

Reply:
Please see 1a1cf0e53bd7741970ba9a465a118a9b45d3c7ef.